Add more information to release notes about event filtering

peterfpeterson opened the pull request:
Adds clarifying comments about arithmetic statistics and how values are deleted from logs.

**To test:**

Review the release notes 

Refs #34794.

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

peterfpeterson committed:
Add more information to release notes about event filtering

sf1919 reviewed (changes requested):
I've looked over the text and that looks ok - although generally the release notes have unnumbered bullet point lists unless there is a very specific reason to number them.

I did build the page and the links work. However there was a warning given that probably needs to be resolved.
`release/v6.7.0/framework:2: WARNING: Bullet list ends without a blank line; unexpected unindent.`

peterfpeterson committed:
Extend header underline to quiet sphinx

peterfpeterson committed:
Change from enumeration to bullet list

peterfpeterson commented:
I changed the enumeration to a bulleted list and fixed a minor issue with a different doc. The warning you point out comes from how the "amalgamate" command works

sf1919 reviewed (approved):
Thanks for changing to a bullet point list. The documentation warnings need to be dealt with but they go beyond just this release note and so I've raised it separately (#35473).
